kvbc/tools/db_editor/include/kv_blockchain_db_editor.hpp

```diff
@@ -0,0 +1,486 @@
+// file.
+
+#pragma once
+
+#include "db_editor_common.hpp"
+#include "categorization/kv_blockchain.h"
+
+namespace concord::kvbc::tools::db_editor {
+
+using namespace categorization;
+
+inline const auto kToolName = "kv_blockchain_db_editor"s;
+inline KeyValueBlockchain getAdapter(const std::string &path, bool read_only = false) {
+  std::cout << "db path " << path << std::endl;
+  auto db = getDBClient(path, read_only);
+
+  // Make sure we don't link the temporary ST chain as we don't want to change the DB in any way.
+  const auto link_temp_st_chain = false;
+  return KeyValueBlockchain{concord::storage::rocksdb::NativeClient::fromIDBClient(db), link_temp_st_chain};
+}
+
+inline BlockId getLatestBlockId(const KeyValueBlockchain &adapter) {
+  auto blockId = adapter.getLastStatetransferBlockId();
+  return blockId ? *blockId : adapter.getLastReachableBlockId();
+}
+
+struct GetGenesisBlockID {
+  bool read_only = true;
+  std::string description() const {
+    return "getGenesisBlockID\n"
+           "  Returns the genesis block ID.";
+  }
+  std::string execute(const KeyValueBlockchain &adapter, const CommandArguments &) const {
+    return toJson("genesisBlockID", adapter.getGenesisBlockId());
+  }
+};
+
+struct GetLastReachableBlockID {
+  bool read_only = true;
+  std::string description() const {
+    return "getLastReachableBlockID\n"
+           "  Returns the last reachable block ID";
+  }
+
+  std::string execute(const KeyValueBlockchain &adapter, const CommandArguments &) const {
+    return toJson("lastReachableBlockID", adapter.getLastReachableBlockId());
+  }
+};
+
+struct GetLastStateTransferBlockID {
+  bool read_only = true;
+  std::string description() const {
+    return "getLastStateTransferBlockID\n"
+           " Returns the last state transfer block ID. If there is no state transfer returns n/a";
+  }
+
+  std::string execute(const KeyValueBlockchain &adapter, const CommandArguments &) const {
+    auto blockId = adapter.getLastStatetransferBlockId();
+    return toJson("lastStateTransferBlockID", blockId ? std::to_string(*blockId) : "n/a"s);
+  }
+};
+
+struct GetLastBlockID {
+  bool read_only = true;
+  std::string description() const {
+    return "getLastBlockID\n"
+           " Returns the last block ID. Either reachable or state transfer";
+  }
+
+  std::string execute(const KeyValueBlockchain &adapter, const CommandArguments &) const {
+    return toJson("lastBlockID", getLatestBlockId(adapter));
+  }
+};
+
+struct GetRawBlock {
+  bool read_only = true;
+  std::string description() const {
+    return "getRawBlock BLOCK-ID\n"
+           "  Returns a serialized raw block (encoded in hex).";
+  }
+
+  std::string execute(const KeyValueBlockchain &adapter, const CommandArguments &args) const {
+    if (args.values.empty()) {
+      throw std::invalid_argument{"Missing BLOCK-ID argument"};
+    }
+    const auto raw_block = adapter.getRawBlock(toBlockId(args.values.front()));
```

Comment:
Maybe can throw an exception if the raw_block is null. Rationale is that the user knows which blocks are available and which are not.